### PR TITLE
Allow multi component CVs

### PR DESCRIPTION
**Pull request by pabloferz**

### CODE REVIEW

File: examples/openmm/metad/nacl/nacl-displacement.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+"""
+Metadynamics simulation of NaCl in water with OpenMM and PySAGES.
+
+Example command to run the simulation `python3 nacl.py --time-steps 1000`
```
Comment (sivadasetty) [inline examples/openmm/metad/nacl/nacl-displacement.py:6]: `nacl.py` to `nacl-displacement.py`

File: examples/openmm/metad/nacl/nacl-displacement.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+For other supported commandline parameters, check `python3 alanine-dipeptide.py --help`
```
Comment (sivadasetty) [inline examples/openmm/metad/nacl/nacl-displacement.py:7]: `alanine-dipeptide.py` to `nacl-displacement.py`

File: examples/openmm/metad/nacl/nacl-displacement.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+import argparse
+import sys
+import time
+
+import numpy
+import pysages
+
+from importlib import import_module
+
+from pysages.colvars import Distance
```
Comment (sivadasetty) [inline examples/openmm/metad/nacl/nacl-displacement.py:23]: Shouldn't this be `Displacement`?

File: examples/openmm/metad/nacl/nacl-displacement.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    for (name, short, T, val, doc) in available_args:
+        parser.add_argument("--" + name, "-" + short, type=T, default=T(val), help=doc)
+    return parser.parse_args(argv)
+
+
+# %%
+def main(argv=[]):
+    args = get_args(argv)
+
+    cvs = [Distance([509, 510])]
```
Comment (sivadasetty) [inline examples/openmm/metad/nacl/nacl-displacement.py:116]: Shouldn't this be `Displacement`?

File: examples/openmm/metad/nacl/nacl-displacement.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+    height = 1.2  # kJ/mol
+    sigma = [0.05]  # nm
+    deltaT = 5000 if args.well_tempered else None
+    stride = 500  # frequency for depositing gaussians
+    timesteps = args.time_steps
+    ngauss = timesteps // stride + 1  # total number of gaussians
+
+    print(args.well_tempered)
```
Comment (sivadasetty) [inline examples/openmm/metad/nacl/nacl-displacement.py:125]: I forgot to delete line 125 and 126.

File: pysages/colvars/core.py [hunk 6]
```diff
@@ -157,7 +182,7 @@ def _get_nargs(function: Callable):
     return len(signature(function).parameters)
 
 
-def _build(cv: CollectiveVariable, grad=jax_grad):
+def _build(cv: CollectiveVariable, diff: bool = True):
```
Comment (sivadasetty) [inline pysages/colvars/core.py:185]: [minor point - can be ignored] diff seems like difference. can we call it `deriv`?

File: pysages/colvars/core.py [hunk 7]
```diff
@@ -182,26 +207,26 @@ def evaluate(positions: JaxArray, ids: JaxArray, **kwargs):
...
-            xi = np.expand_dims(function(pos, ids, **kwargs).flatten(), 0)
-            Jxi = np.expand_dims(gradient(pos, ids, **kwargs).flatten(), 0)
-            return xi, Jxi
+            return cv_fn(pos, ids, **kwargs).reshape(1, -1)
 
     return jit(apply)
 
 
-def build(cv: CollectiveVariable, *cvs: CollectiveVariable, grad=jax_grad):
+def build(cv: CollectiveVariable, *cvs: CollectiveVariable, diff: bool = True):
```
Comment (sivadasetty) [inline pysages/colvars/core.py:229]: same as above for `diff`.

File: pysages/colvars/core.py [hunk 8]
```diff
@@ -182,26 +207,26 @@ def evaluate(positions: JaxArray, ids: JaxArray, **kwargs):
             pos = [pos[g] for g in gps]
             return np.asarray(xi(*pos, **kwargs))
 
-    function = jit(evaluate)
+    cv_fn = jit(evaluate)
 
-    if grad is None:
+    if diff:
```
Comment (sivadasetty) [inline pysages/colvars/core.py:212]: same point as above for `diff`.

File: pysages/colvars/core.py (hunk 6 above)
Comment (pabloferz) [reply pysages/colvars/core.py]: Change it to `differentiate`.

**Review by sivadasetty — APPROVED** [review]

**Pull request merged**